test/tests/services/fields-flattener.test.js

```diff
@@ -0,0 +1,197 @@
+            const fieldsFlattener = new FieldsFlattener({ fields: [] }, [{ field: 'test' }]);
+            fieldsFlattener.validateOptions();
+
+            expect(warnLoggerSpy).toHaveBeenCalledTimes(1);
+          });
+
+          it('should remove the non existing field from fields to flatten', () => {
+            expect.assertions(1);
+
+            const fieldsFlattener = new FieldsFlattener({ fields: [] }, [{ field: 'test' }]);
+            fieldsFlattener.validateOptions();
+
+            expect(fieldsFlattener.flatten).toHaveLength(0);
+          });
+        });
+      });
+
+      describe('when a field has not been specified in flatten object', () => {
+        it('should display a warning message', () => {
+          expect.assertions(1);
+
+          jest.resetAllMocks();
+          const fieldsFlattener = new FieldsFlattener({ fields: [] }, [{ }]);
+          fieldsFlattener.validateOptions();
+
+          expect(warnLoggerSpy).toHaveBeenCalledTimes(1);
+        });
+
+        it('should remove the non existing field from fields to flatten', () => {
+          expect.assertions(1);
+
+          const fieldsFlattener = new FieldsFlattener({ fields: [] }, [{ }]);
+          fieldsFlattener.validateOptions();
+
+          expect(fieldsFlattener.flatten).toHaveLength(0);
+        });
+      });
+    });
+  });
+
+  describe('flattening a field', () => {
+    it('should merge fields name with | as separator', () => {
+      expect.assertions(1);
+
+      const schema = generateEngineSchema();
+
+      const fieldsFlattener = new FieldsFlattener(schema, ['engine']);
+      fieldsFlattener.flattenFields();
+
+      expect(schema.fields[0].field).toContain('|');
+    });
+  });
+
+  describe('when the level property is empty', () => {
+    it('should flatten every nested fields until the end', () => {
+      expect.assertions(1);
+
+      const schema = generateEngineSchema();
+      const fieldsFlattener = new FieldsFlattener(schema, ['engine']);
+
+      fieldsFlattener.flattenFields();
+
+      expect(schema.fields).toHaveLength(5);
+    });
+  });
+
+  describe('when the level property is specified', () => {
+    describe('when the level property is above 0', () => {
+      describe('when the level property is higher than actual level', () => {
+        it('should flatten every level until the end', () => {
+          expect.assertions(1);
+
+          const schema = generateEngineSchema();
+          const fieldsFlattener = new FieldsFlattener(schema, [{ field: 'engine', level: 100 }]);
+
+          fieldsFlattener.flattenFields();
+
+          expect(schema.fields).toHaveLength(5);
+        });
+      });
+      describe('when the level property is lower than actual level', () => {
+        it('should flatten only until specified level', () => {
+          expect.assertions(1);
+
+          const schema = generateEngineSchema();
+          const fieldsFlattener = new FieldsFlattener(schema, [{ field: 'engine', level: 0 }]);
```

Comment:
maybe rather test with `level: 1` to not repeat the following test?

Reply:
Indeed, as a consequence I have updated the data set for the test suit.